mm2src/lp_swap/taker_swap.rs

```diff
@@ -1323,13 +1390,34 @@ impl TakerSwap {
 
         let transaction = match refund_fut.compat().await {
             Ok(t) => t,
-            Err(err) => {
-                return Ok((Some(TakerSwapCommand::Finish), vec![
-                    TakerSwapEvent::TakerPaymentRefundFailed(ERRL!("{}", err).into()),
-                ]))
+            Err(err_enum) => match err_enum {
```

Comment:
Can we have a more specific type here or at least a helper fn that will work like
```rust
if let Some(tx) = tx_send_result.get_tx()/get_tx(&tx_send_result) {
   broadcast_p2p_tx_msg(...)
}

if let Err(e) = tx_send_result {
   return Ok((Some(TakerSwapCommand::Finish) ...
}
```

Reply:
I am also thinking to refactor this. I will take a look at it once I finish the tests.